### ISSUE TITLE
Train rasa_core with dialogflow files

### ISSUE THREAD

**Issue by Priprocks** [RasaHQ/rasa_core#508]
**Rasa Core version**:
0.9.a3
**Python version**: 
3.6
**Operating system** (windows, osx, ...):
Windows 10
**Issue**:
Is there any way I could use the dialogflow files to train the rasa_core dialog model as well?
I mean I am using these files to train the rasa_nlu model and it seems to work fine.

**Content of domain file** (if used & relevant):
```yaml

```

**Comment by tmbo** [comment]
At the moment there is no converter for that data to the rasa core format.

**Comment by tmbo** [comment]
Sure, I think the hardest part is reading yourself into the dialogue flow export format. So what you'd need to do is:
- write a script that takes a folder as a parameter 
- uses Rasa NLU converter to create the NLU training data
- figure out how to create the stories and domain file from the remaining information you get from dialogflow

in the end there should be a file for the nlu training, a story file and a domain file.

**Comment by Priprocks** [comment]
One more thing, I am new to making chatbots in rasa. So, is there any slack workspace or something where I can clear my queries?
Thanks for the help :)

**Comment by tmbo** [comment]
@Priprocks how are things going, do you need any help?

I think for the if else parts, it would be good to create stories that mimic that - so kind of example stories for the different branches. does that make sense?

**Comment by Priprocks** [comment]
@tmbo Actually I never contributed to RASA as such. So, I was thinking to document some of the code, make PR for them and then get back to it. Coding this directly seems quite a challenge.

**Comment by bhavaniravi** [comment]
@tmbo I am trying to work on this feature as well.
Does the order of the story in stories file matters?

**Comment by tmbo** [comment]
No, the order does not matter

**Comment by bhavaniravi** [comment]
@tmbo I have built it as a separate script. Needs some testing. Once done I would need some help on how to go about plugging it in to the existing codebase

https://github.com/bhavaniravi/dialogflow_to_rasa_core

**Comment by Priprocks** [comment]
I wrote a script to make the domain file as of now.
Will send a PR once my current one gets accepted/rejected.

**Comment by tmbo** [comment]
I think these are two things, one is to get a bot from dialogflow and convert & train it with rasa core. the second part is to run the rasa core server to emulate the API of dialogflow.

I think the second one is not as important, and the conversion of the dialogflow data to a rasa core bot should be done first.

**Comment by Priprocks** [comment]
I submitted the PR for the 2nd part a few weeks back.
For the first part, I wrote a script that's successfully generating the domain.yml file.
Although, there are a few cases that I couldn't handle.
Also, I couldn't think of a way to make the story file too.

**Comment by tmbo** [comment]
@bhavaniravi really cool work. would be good to combine these efforts to get a good converter. @bhavaniravi did you already do some tests of the conversion?

**Comment by bhavaniravi** [comment]
@priprocks @tmbo I can take care of the stories part

**Comment by bhavaniravi** [comment]
@tmbo yes I have tested it with couple of dialogflow built in agents and stories also works fine. Needs unit testing.

**Comment by Priprocks** [comment]
I'll test it then on my dialogflow exports. Isn't making a branch of the master better. Let's work on that separately and when all will seem fit, we'll create the PR.

**Comment by bhavaniravi** [comment]
@Priprocks  Sure lets do it.

**Comment by Priprocks** [comment]
Okay, I'm currently going through the code. Will further the communication in your repo.

**Comment by anshulsolanki** [comment]
Please keep me also in loop, i can test. I did test @bhavaniravi first version. It worked good for NLU part , but for stories as i tested more, there is still more work to be done.

**Comment by Priprocks** [comment]
@tmbo Should we go ahead with the repo of @bhavaniravi ?
I mean should this work as an independent project or should it be merged here?

**Comment by Priprocks** [comment]
So, I had a slightly different take on making the files than @bhavaniravi 
I have put it up [here](https://github.com/Priprocks/Dialogflow_export_to_rasa).
Checked with my export and the prebuilt agents of Dialogflow and its working fine for them.
Haven't yet dealt with contexts so that's there still but it won't affect the output.
@anshulsolanki  if you have the time, then do test it.

**Comment by Priprocks** [comment]
1) Haven't handled the case of NLU builder. Its provided by rasa_nlu so I used that.
`python driver.py train-nlu`
2) Fixed the action issue in the latest commit.
3) I really didn't think about integrating.

I'll explain the flow:
For each action, I'm making 2 action commands: 
- custom action script
- utterance

Now, the custom action will handle the issues for outgoing context and required parameters and then call the dispatcher to utter its template.

The context portion is messed up still.

**Comment by anshulsolanki** [comment]
OK thanks, do you plan to work on them?

**Comment by Priprocks** [comment]
I couldn't think of a way to deal with the context part better. If you have any idea on how should this be handled then let me know.

**Comment by Priprocks** [comment]
Also, there's the issue with required parameters.
So, what happens is, if I set a required parameter with a prompt, in dialogflow, it never leaves the parent intent but in rasa, it misclassifies the intent.
For eg.,
Suppose there's an intent `bug_report` with required parameter: `bug_domain`
Now, if its missing in the text of a user when the text gets classified to bug_report intent, it prompts with user asking it to specify the `bug domain`.
Now the user can say "booking movie".
`bug_domain` entity is filled up with booking movie and the response of bug_report is uttered.
This is how it works in Dialogflow.
In Rasa, if the user says "booking movie", it won't stay in the `bug_report` intent.
Rather, it would classify the text as "movie_booking" intent and the whole flow gets broken.
